--- origsrc/tcl8.4.19/generic/tclLoad.c	2003-02-01 17:37:29.000000000 -0600
+++ src/tcl8.4.19/generic/tclLoad.c	2008-07-17 01:08:17.109375000 -0500
@@ -292,6 +292,12 @@
 			&& (pkgGuess[2] == 'b')) {
 		    pkgGuess += 3;
 		}
+#ifdef __CYGWIN__
+		else if ((pkgGuess[0] == 'c') && (pkgGuess[1] == 'y')
+			&& (pkgGuess[2] == 'g')) {
+		    pkgGuess += 3;
+		}
+#endif
 		for (p = pkgGuess; *p != 0; p += offset) {
 		    offset = Tcl_UtfToUniChar(p, &ch);
 		    if ((ch > 0x100)
--- origsrc/tcl8.4.19/unix/configure.in	2008-04-17 15:29:49.000000000 -0500
+++ src/tcl8.4.19/unix/configure.in	2008-07-17 01:08:17.109375000 -0500
@@ -561,7 +561,7 @@
 
 TCL_UNSHARED_LIB_SUFFIX=${UNSHARED_LIB_SUFFIX}
 TCL_SHARED_LIB_SUFFIX=${SHARED_LIB_SUFFIX}
-eval "TCL_LIB_FILE=libtcl${LIB_SUFFIX}"
+eval "TCL_LIB_FILE=cygtcl${LIB_SUFFIX}"
 
 # tclConfig.sh needs a version of the _LIB_SUFFIX that has been eval'ed
 # so that the backslashes quoting the DBX braces are dropped.
@@ -645,8 +645,8 @@
         eval "TCL_EXP_FILE=\"${TCL_EXP_FILE}\""
         
         if test "$GCC" = "yes" ; then
-            TCL_BUILD_LIB_SPEC="-Wl,-bI:`pwd`/${TCL_BUILD_EXP_FILE} -L`pwd`"
-            TCL_LIB_SPEC="-Wl,-bI:${libdir}/${TCL_EXP_FILE} -L`pwd`"
+            TCL_BUILD_LIB_SPEC="-Wl,`pwd`/${TCL_BUILD_EXP_FILE} -L`pwd`"
+            TCL_LIB_SPEC="-L${libdir} -ltcl${TCL_VERSION}\${TCL_DBGX}"
         else
             TCL_BUILD_LIB_SPEC="-bI:`pwd`/${TCL_BUILD_EXP_FILE}"
             TCL_LIB_SPEC="-bI:${libdir}/${TCL_EXP_FILE}"
--- origsrc/tcl8.4.19/unix/installManPage	2008-04-17 15:29:49.000000000 -0500
+++ src/tcl8.4.19/unix/installManPage	2008-07-17 01:08:17.125000000 -0500
@@ -3,7 +3,7 @@
 ZIP=:
 while true; do
     case $1 in
-        -s | --symlinks  )      S="-s ";;
+        -s | --symlinks  )      SYM="-s ";;
         -z | --compress  )    ZIP=$2;  shift ;;
 	-e | --extension )      Z=$2;  shift ;;
 	-s | --suffix    ) SUFFIX=$2;  shift ;;
@@ -18,7 +18,7 @@
 
 MANPAGE=$1
 DIR=$2
-test -z "$S" && S="$DIR/"
+test -z "$SYM" && SYM="$DIR/"
 
 # A sed script to parse the alternative names out of a man page.
 #
@@ -58,10 +58,10 @@
 	rm -f $DIR/$FIRST $DIR/$FIRST.*
 	sed -e "/man\.macros/r $SRCDIR/man.macros" -e "/man\.macros/d" \
 	    $MANPAGE > $DIR/$FIRST
-	chmod 444 $DIR/$FIRST
+	chmod 644 $DIR/$FIRST
 	$ZIP $DIR/$FIRST
     else
 	rm -f $DIR/$f $DIR/$f.*
-	ln $S$FIRST$Z $DIR/$f$Z
+	ln $SYM$FIRST$Z $DIR/$f$Z
     fi
 done
--- origsrc/tcl8.4.19/unix/tcl.m4	2008-04-17 15:29:49.000000000 -0500
+++ src/tcl8.4.19/unix/tcl.m4	2008-07-17 01:08:17.140625000 -0500
@@ -1182,6 +1182,23 @@
 	    CC_SEARCH_FLAGS=""
 	    LD_SEARCH_FLAGS=""
 	    ;;
+	CYGWIN*)
+		SHLIB_CFLAGS=""
+		SHLIB_LD='${CC} -shared -Wl,--enable-auto-image-base'
+		TCL_SHLIB_LD_EXTRAS='-Wl,--out-implib,${TCL_BUILD_EXP_FILE}'
+		SHLIB_LD_LIBS='${LIBS}'
+		SHLIB_SUFFIX=".dll"
+	    CFLAGS_OPTIMIZE=-O2
+		CC_SEARCH_FLAGS=""
+		LD_SEARCH_FLAGS=""
+		DL_OBJS="tclLoadDl.o"
+		DL_LIBS=""
+		LDFLAGS="$LDFLAGS -Wl,--export-dynamic"
+		TCL_NEEDS_EXP_FILE=1
+		TCL_EXPORT_FILE_SUFFIX='${VERSION}\$\{DBGX\}.dll.a'
+	    AC_DEFINE(PEEK_XCLOSEIM)
+		AC_DEFINE(HAVE_TM_ZONE)
+	    ;;
 	dgux*)
 	    SHLIB_CFLAGS="-K PIC"
 	    SHLIB_LD="cc -G"
@@ -2089,6 +2106,8 @@
 		    ;;
 		BSD/OS*)
 		    ;;
+		CYGWIN*)
+			;;
 		IRIX*)
 		    ;;
 		NetBSD-*|FreeBSD-*|OpenBSD-*)
