--- origsrc/tcl8.5.11/generic/tclEnv.c	2013-03-07 21:56:14.713821700 -0600
+++ src/tcl8.5.11/generic/tclEnv.c	2013-03-07 21:57:15.374291200 -0600
@@ -46,8 +46,7 @@ MODULE_SCOPE void	TclUnsetEnv(const char
 
 #if defined(__CYGWIN__)
 /* On Cygwin, the environment is imported from the Cygwin DLL. */
-     DLLIMPORT extern int cygwin_posix_to_win32_path_list_buf_size(char *value);
-     DLLIMPORT extern void cygwin_posix_to_win32_path_list(char *buf, char *value);
+#include <sys/cygwin.h>
 #    define putenv TclCygwinPutenv
 static void		TclCygwinPutenv(char *string);
 #endif
@@ -779,9 +778,9 @@ TclCygwinPutenv(
 	} else {
 	    int size;
 
-	    size = cygwin_posix_to_win32_path_list_buf_size(value);
-	    buf = alloca(size + 1);
-	    cygwin_posix_to_win32_path_list(value, buf);
+	    size = cygwin_conv_path_list(CCP_POSIX_TO_WIN_A, value, NULL, 0);
+	    buf = alloca(size);
+	    cygwin_conv_path_list(CCP_POSIX_TO_WIN_A, value, buf, size);
 	}
 
 #ifdef __WIN32__
--- origsrc/tcl8.5.11/generic/tclPort.h	2011-11-01 09:04:51.000000000 -0500
+++ src/tcl8.5.11/generic/tclPort.h	2013-03-07 21:56:46.748653900 -0600
@@ -28,11 +28,6 @@
 #if defined(__CYGWIN__)
 #   define USE_PUTENV 1
 #   define USE_PUTENV_FOR_UNSET 1
-/* On Cygwin, the environment is imported from the Cygwin DLL. */
-    DLLIMPORT extern char **__cygwin_environ;
-    DLLIMPORT extern int cygwin_conv_to_win32_path(const char *, char *);
-#   define environ __cygwin_environ
-#   define timezone _timezone
 #endif
 
 #if !defined(LLONG_MIN)
--- origsrc/tcl8.5.9/generic/tclLoad.c	2008-11-13 18:22:39.000000000 -0600
+++ src/tcl8.5.9/generic/tclLoad.c	2011-02-15 01:34:56.716143000 -0600
@@ -310,6 +310,12 @@ Tcl_LoadObjCmd(
 			&& (pkgGuess[2] == 'b')) {
 		    pkgGuess += 3;
 		}
+#ifdef __CYGWIN__
+		if ((pkgGuess[0] == 'c') && (pkgGuess[1] == 'y')
+			&& (pkgGuess[2] == 'g')) {
+		    pkgGuess += 3;
+		}
+#endif
 		for (p = pkgGuess; *p != 0; p += offset) {
 		    offset = Tcl_UtfToUniChar(p, &ch);
 		    if ((ch > 0x100)
--- origsrc/tcl8.5.9/unix/tcl.m4	2010-09-08 12:40:48.000000000 -0500
+++ src/tcl8.5.9/unix/tcl.m4	2011-02-15 01:43:22.973429500 -0600
@@ -1257,6 +1257,9 @@ dnl AC_CHECK_TOOL(AR, ar)
 	    DL_LIBS="-ldl"
 	    CC_SEARCH_FLAGS=""
 	    LD_SEARCH_FLAGS=""
+	    TCL_NEEDS_EXP_FILE=1
+	    TCL_EXPORT_FILE_SUFFIX='${VERSION}\$\{DBGX\}.dll.a'
+	    TCL_SHLIB_LD_EXTRAS='-Wl,--out-implib,$[@].a'
 	    ;;
 	dgux*)
 	    SHLIB_CFLAGS="-K PIC"
